packages/squiggle-lang/src/value/ValueTagsWrapper.ts

```diff
@@ -0,0 +1,85 @@
+import { Location, LocationRange } from "peggy";
+
+import { ImmutableMap } from "../utility/immutableMap.js";
+import { Value, vNumber, vString } from "./index.js";
+import { ValueTags } from "./valueTags.js";
+import { vDict, type VDict } from "./VDict.js";
+
+function locationRangeToValue(locationRange: LocationRange) {
+  function locationToDict(location: Location): VDict {
+    return vDict(
+      ImmutableMap([
+        ["line", vNumber(location.line)],
+        ["column", vNumber(location.column)],
+        ["offset", vNumber(location.offset)],
+      ])
+    );
+  }
+  let items: [string, Value][] = [
+    ["start", locationToDict(locationRange.start)],
+    ["end", locationToDict(locationRange.end)],
+  ];
+  if (typeof locationRange.source === "string") {
+    items = [["source", vString(locationRange.source)], ...items];
+  }
+  return vDict(ImmutableMap(items));
+}
+
+// This class is meant for functions that depend on Value. BaseValue depends on ValueTags, so we can't call many functions there. Instead, we call them here.
+export class ValueTagsWrapper {
```

Comment:
I think you could safely skip OOP here and use a module-as-plain-object:

```typescript
export const tagUtils = {
  toVDict: (tags: ValueTags) => { ... },
  toValueMap: (tags: ValueTags) => { ... },
  toLocation: (tags: ValueTags) => { ... },
}
```

And then `return tagUtils.toValueMap(tags)`.

(AFAICT the only function you need here is `toValueMap`, but exporting many small functions one by one can be annoying and less readable, so a top-level object is more convenient)